dali/pipeline/operators/transpose/transpose.h

```diff
@@ -24,36 +24,74 @@
 
 namespace dali {
 
+namespace detail {
+
+template <typename T>
+T Permute(const T& in, const std::vector<int>& permutation) {
+  T out = in;
+  for (size_t i = 0; i < permutation.size(); i++) {
+    auto idx = permutation[i];
+    out[i] = in[idx];
+  }
+  return out;
+}
+
+}  // namespace detail
+
+
 template <typename Backend>
 class Transpose : public Operator<Backend> {
  public:
-  explicit inline Transpose(const OpSpec &spec) :
-    Operator<Backend>(spec),
-    perm_(spec.GetRepeatedArgument<int>("perm")) {
-      DALI_ENFORCE([](std::vector<int> perm) {
-          std::sort(perm.begin(), perm.end());
-          for (int i = 0; i < static_cast<int>(perm.size()); ++i) {
-            if (perm[i] != i) {
-              return false;
-            }
+  explicit inline Transpose(const OpSpec &spec)
+      : Operator<Backend>(spec)
+      , perm_(spec.GetRepeatedArgument<int>("perm"))
+      , transpose_layout_(spec.GetArgument<bool>("transpose_layout"))
+      , output_layout_arg_(spec.GetArgument<TensorLayout>("output_layout")) {
+    if (spec.HasArgument("output_layout")) {
+      DALI_ENFORCE(!output_layout_arg_.empty(),
+        "Providing an empty output layout is not supported");
+    }
+
+    DALI_ENFORCE(
+      [](std::vector<int> perm) {
+        std::sort(perm.begin(), perm.end());
+        for (int i = 0; i < static_cast<int>(perm.size()); ++i) {
+          if (perm[i] != i) {
+            return false;
           }
-          return true;
-        }(perm_), "Invalid permutation: sorted `perm` is not equal to [0, ..., n-1].");
+        }
+        return true;
+      }(perm_), "Invalid permutation: sorted `perm` is not equal to [0, ..., n-1].");
```

Comment:
This will put the whole loop in the error message.
How about declaring a variable (like `bool is_valid_perm`) and enforce that?